### DISCUSSION TITLE
basic question on merging and also on reversion to particular commit state

### DISCUSSION THREAD

**Discussion by dmastrop** [community/community#53365] [Programming Help] [answered]
### Body

hi all

I have a basic question on merging.
There is a development branch and a master branch.
the development branch was not worked on much and has some old scripts in it.

I technically should have been working in the development branch, but the scripts are written to deploy only on master branch and I wanted to test it out very quickly. So I did all the changes on master branch.

The scripts were made to work and I wanted to "synch". (Make the same) the development branch as the master branch.

For example, 
it initially looked like this
/github/workflows/
and in this directory a deploy1.yaml and a frontend.yaml file.

The master branch has a
/.github/worlflows of
deploy2.yaml and frontend.yaml

The deploy2.yaml is the one that actually works in deployment and I wanted this copy in the development branch.

I merged the master into the development (at remote side) and ended up with 

% ls
deploy1.yaml            deploy2_wif.yaml        frontend.yaml

but the deploy1.yaml was moved out of the directory in the master branch prior to the merge.

I can understand why the merge would leave the deploy1.yaml script in the /.github/workflows directory but what command should I have used to synch up and make the development branch the exact same as the master branch.?

Should I have manually added the commits to the development branch to do this?

Note: I don't want the deploy1.yaml in the workflows directory any longer as it will break the deployment, and even though the development branch won't be deployed I wanted to synch the changes in master over to the development branch.

==========

For reversion of this in development branch, I tried git revert with the commit that I wanted to go to and that just undoes the particular commit.
The **git reset** actually brought me back to the state of that commit which is what I wanted.

"What is the difference between git revert and reset?
While git reset does this by moving the current head of the branch back to the specified commit, thereby changing the commit history, git revert does this by creating a new commit that undoes the changes in the specified commit and so does not change the history."

To rewind to a particular commit state back in history is the git reset the proper command?

==========

Once back in that state I did notice the untracked added file hanging there (shown in git status) and still in the directory.  Do I need to manually delete the file or is there a git command that I can use to revert the Untracked change (added file)?
It looks like everything reverted and this file is rightfully untracked.  I ended up manually deleting it per the directions when I tried to checkout.

% git status
On branch development
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        deploy2_wif.yaml

nothing added to commit but untracked files present (use "git add" to track)

 % git checkout master
error: The following untracked working tree files would be overwritten by checkout:
        .github/workflows/deploy2_wif.yaml
Please move or remove them before you switch branches.
Aborting

**removed the file**
 

% git status
On branch development
nothing to commit, working tree clean

% git checkout master

warm regards.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by antonblr** [comment, marked as answer]
Hi @dmastrop,

1) You could reset development branch to `master` locally and then force-push it to the remote. Something like:
```shell
git checkout development
git reset --hard master  # assuming your local "master" branch is up-to-date
git push origin development --force
```

2) Yes,  [git reset](https://git-scm.com/docs/git-reset) is more appropriate for this.
3) You could use [git clean](https://git-scm.com/docs/git-clean) to remove untracked files (add `-d` to include directories):
```shell=
git clean --dry-run # dry mode to see what will be removed
git clean -f # to actually remove files
```

**Reply by dmastrop** [reply]
Many thanks Anton. This worked really well. 
And the git clean is exactly what I was looking for.  I ended up manually removing but the git clean will be much better for the future.
Warm regards
